uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -363,11 +376,11 @@ define([
       function initOpenMode() {
         if (vm.leaveRequest.id) {
           vm.mode = 'edit';
-          //todo in future
-          //if owner and status is approved then view only mode
-          // if (vm.role === 'owner' && vm.leaveRequest.status_id == valueOfRequestStatus('approved')){
-          //   vm.mode = 'view';
-          // }
+
+          var viewModes = ['1', '2', '5', '6'];//approved, admin_approved, rejected, cancelled
```

Comment:
We should not hardcode values, instead we should `find` values for each type from `status` array.
```javascript
var viewModes = [_.find(... 'approved'), _.find(... 'admin_approved', ...)];